integritybackend/claim.py

```diff
@@ -97,19 +97,63 @@ def generate_create(self, jwt_payload, data):
 
         return claim
 
-    def generate_create_proofmode(self, jwt_payload, meta_proofmode):
-        """Generates a claim for the 'create_proofmode' action.
+    def generate_c2pa_proofmode(self, action_params: dict, meta_content: dict, filename: str):
+        """Generates a claim for the 'c2pa-proofmode' action.
 
         Args:
-            jwt_payload: a dictionary with the data we got from the request's JWT payload
-            meta_proofmode: dictionary with the metadata from a proofmode bundle
+            action_params: a dictionary with the params from this action's config
+            meta_content: dictionary with the metadata from a proofmode bundle
+            filename: 
 
         Returns:
             a dictionary containing the 'create' claim data
         """
         claim = copy.deepcopy(CREATE_CLAIM_TEMPLATE)
+        claim["recorder"] = "ProofMode by Guardian Project and WITNESS"
+
+        assertion_templates = self.assertions_by_label(claim)
+        assertions = []
+
+        author_data = action_params["creative_work_author"]
+        if author_data is not None:
+            creative_work = assertion_templates["stds.schema-org.CreativeWork"]
+            creative_work["data"] = author_data
+            assertions.append(creative_work)
+
+        author_name = action_params["creative_work_author"]["name"]
+        copyright = action_params["copyright"]
+        gps_lat = meta_content["private"]["proofmode"][filename]["proofs"][0]["Location.Latitude"]
+        gps_lon = meta_content["private"]["proofmode"][filename]["proofs"][0]["Location.Longitude"]
+        photo_meta_data = self._make_photo_meta_data(author_name, copyright, gps_lat, gps_lon)
```

Comment:
@YurkoWasHere what happens if we're without GPS 🤔

Reply:
I think you may want to prioritize the GPS in the exif first, then proof mode? Not sure

In the context of SCMP sounds like they will have picture-taken gps in exif while proof mode will be prooved gps coords not taken.

If empty maybe set to 0 0  or not include it at all, not sure.

Reply:
Not using exif